airbyte-cdk/python/airbyte_cdk/sources/streams/core.py

```diff
@@ -441,6 +458,53 @@ def _checkpoint_mode(self) -> CheckpointMode:
         else:
             return CheckpointMode.FULL_REFRESH
 
+    @staticmethod
+    def _detect_stream_category(mappings_or_slices: Iterable[Union[Mapping[str, Any], StreamSlice]]) -> tuple[bool, bool]:
```

Comment:
nit: can this method return a dataclass with two fields for clarity?

```
@dataclass
class StreamCategory:
    is_legacy_format: bool
    is_substream: bool
```

Reply:
done!